src/pages/index.js

```diff
@@ -100,15 +94,163 @@ const cardSection = new Section(
+const imagePopup = new PopupWithImage("#preview-card-modal");
+imagePopup.setEventListeners();
+
+const profileFormPopup = new PopupWithForm("#edit-profile-modal", handleProfileFormSubmit);
+profileFormPopup.setEventListeners();
+
+const addCardFormPopup = new PopupWithForm("#add-card-modal", handleAddCardFormSubmit);
+addCardFormPopup.setEventListeners();
+
+const deleteConfirm = new PopupWithConfirm("#delete-card-modal");
+deleteConfirm.setEventListeners();
+
+const avatarPopup = new PopupWithForm("#edit-avatar-modal", handleAvatarFormSubmit);
+avatarPopup.setEventListeners();
+
+/* -------------------------------------------------------------------------- */
+/*                               Event Handlers                               */
+/* -------------------------------------------------------------------------- */
+function handleSubmit(request, popupInstance, loadingText = "Saving...") {
+  popupInstance.renderLoading(true, loadingText);
+  request()
+    .then(() => {
+      popupInstance.close();
+    })
+    .catch((error) => {
+      console.error("Error:", error);
+    })
+    .finally(() => {
+      popupInstance.renderLoading(false);
+    });
 }
 
 function handleImageClick(name, link) {
-    imagePopup.open(name, link);
+  imagePopup.open(name, link);
 }
 
+function handleProfileFormSubmit(inputValues) {
+  function makeRequest() {
+    return api.updateUserInfo({
+      name: inputValues.name,
+      job: inputValues.description,
+    }).then((userData) => {
+      userInfo.setUserInfo({
+        name: userData.name,
+        job: userData.about,
+      });
+    });
+  }
+  handleSubmit(makeRequest, profileFormPopup);
+}
 
+const cardForm = document.querySelector("#add-card-form");
+function handleAddCardFormSubmit(inputValues) {
+  function makeRequest() {
+    const cardData = { title: inputValues.title, url: inputValues.url };
+    return api.addCard(cardData).then((savedCardData) => {
+      const card = createCard({
+        name: savedCardData.name,
+        link: savedCardData.link,
+        _id: savedCardData._id,
+      });
+      cardSection.addItem(card);
+      cardForm.reset();
+    });
+  }
+  handleSubmit(makeRequest, addCardFormPopup, "Creating...");
+}
 
+function handleDeleteCard(card) {
+  deleteConfirm.setSubmitFunction(() => {
+    api.deleteCard(card._id)
+      .then(() => {
+        console.log(`Card ${card._id} deleted successfully`);
+        card.handleDeleteCard();
+      })
+      .catch((error) => {
+        console.error("Error deleting card:", error);
+      });
+  });
+  deleteConfirm.open();
+}
+
+const avatarForm = document.querySelector("#edit-avatar-form");
+function handleAvatarFormSubmit(inputValues) {
+  function makeRequest() {
+    return api.updateUserAvatar(inputValues.avatarUrl).then((userData) => {
+      userInfo.setUserInfo({ avatar: userData.avatar });
+    });
+  }
+  handleSubmit(makeRequest, avatarPopup);
+  avatarForm.reset();
```

Comment:
`  avatarForm.reset();
`


You need to **reset** the form only   in a `then` block in order not to do that immediately after `submit` because when there is an error, the modal will still be opened and the user can try to send the data again to the server without having to type the data into the inputs once more.